core/src/network/p2p.rs

```diff
@@ -298,8 +301,17 @@ async fn build_swarm(
 
 	let kad_cfg: kad::Config = kad_config(cfg, genesis_hash);
 
+	let connection_limits = ConnectionLimits::default()
+		.with_max_pending_incoming(Some(100))
```

Comment:
How did we arrive to these numbers?

Reply:
Completely arbitrary.
Shouldn't be hard coded IMHO.